app/packages/app/src/useEvents/useRefresh.ts

```diff
@@ -1,8 +1,37 @@
-import { useRefresh as useRefreshState } from "@fiftyone/state";
+import { subscribe } from "@fiftyone/relay";
+import * as fos from "@fiftyone/state";
+import { env } from "@fiftyone/utilities";
+import { useCallback } from "react";
+import { getDatasetName, getParam, resolveURL } from "../utils";
 import { EventHandlerHook } from "./registerEvent";
+import { processState } from "./utils";
 
-const useRefresh: EventHandlerHook = () => {
-  return useRefreshState();
+const useRefresh: EventHandlerHook = ({ router, session }) => {
+  return useCallback(
+    (payload: any) => {
```

Comment:
Suggestion, feel free to ignore:

- extract this function
- add a unit test
- optional: ideally it would fail before this change